src/runners/ActualRunner.cpp

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2018 DKFZ - ODCF
+ *
+ * Distributed under the MIT License (license terms are at https://github.com/dkfz-odcf/FastqIndEx/blob/master/LICENSE.txt).
+ */
+
+#include "common/ErrorMessages.h"
+#include "process/io/StreamSource.h"
+#include "runners/ActualRunner.h"
+#include <experimental/filesystem>
+
+using experimental::filesystem::path;
+
+ActualRunner::ActualRunner(const shared_ptr<Source> &fastqfile) {
+    this->fastqFile = fastqfile;
+}
+
+bool ActualRunner::fulfillsPremises() {
+
+    // Fastq needs to be a (pipe AND piping allowed) or an ((existing file OR symlink with a file) AND readable)
```

Comment:
Instead of this somewhat cryptic comment, why not put the pipe-condition into the `else if (...)` and additionally have an `else` -- that probably never is called, but might get relevant in some programming bug conditions.

The advantage would be that everything is explicit and in the code -- which I'd prefer much over this comment.